### ISSUE TITLE
`$ORIGIN` in RPATH on macOS

### ISSUE THREAD

**Issue by alexanderlobov** [bazelbuild/bazel#4480]
### Description of the problem / feature request:

A binary built by bazel has linux-specific `$ORIGIN` variable in RPATH on macOS.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

[Minimal example here](https://github.com/alexanderlobov/try-bazel/tree/master/linking-with-dynamic-lib).

To build:

```
alex@osx:~/src/github/try-bazel/linking-with-dynamic-lib$ /usr/local/bin/bazel build app
INFO: Analysed target //app:app (9 packages loaded).
INFO: Found 1 target...
Target //app:app up-to-date:
  bazel-bin/app/app
INFO: Elapsed time: 1.191s, Critical Path: 0.03s
INFO: Build completed successfully, 1 total action
```

To check RPATH

```
alex@osx:~/src/github/try-bazel/linking-with-dynamic-lib$ otool -l bazel-bin/app/app | grep -A2 RPATH
          cmd LC_RPATH
      cmdsize 64
         path $ORIGIN/../_solib_darwin/_U_S_Slib_Clib___Ulib (offset 12)
```

As a result of wrong RPATH the binary can not start:

```
alex@osx:~/src/github/try-bazel/linking-with-dynamic-lib$ bazel-bin/app/app
dyld: Library not loaded: @rpath/libpython3.6m.dylib
  Referenced from: /Users/alex/src/github/try-bazel/linking-with-dynamic-lib/bazel-bin/app/app
  Reason: image not found
Abort trap: 6
```

I have tried `cc_import` instead of `cc_library`, but it fails too. To do this, comment the `cc_library` in `lib/BUILD.bazel` and uncomment `cc_import` declaration.

```
alex@osx:~/src/github/try-bazel/linking-with-dynamic-lib$ /usr/local/bin/bazel build app
ERROR: /Users/alex/src/github/try-bazel/linking-with-dynamic-lib/app/BUILD.bazel:4:12: in deps attribute of cc_binary rule //app:app: cc_import rule '//lib:lib' is misplaced here (expected cc_inc_library, cc_library, objc_library or cc_proto_library)
ERROR: Analysis of target '//app:app' failed; build aborted: Analysis of target '//app:app' failed; build aborted
INFO: Elapsed time: 0.149s
FAILED: Build did NOT complete successfully (1 packages loaded)
```

At the same time, `cc_import` seems to work on Linux.

I also tried the development version of bazel (from master branch), but it fails to link with libstdc++, see https://github.com/bazelbuild/bazel/issues/4479 for details.

### What operating system are you running Bazel on?

macOS Sierra 10.12.6

### What's the output of `bazel info release`?

```
release 0.9.0-homebrew
```

###  Have you found anything relevant by searching the web?

https://github.com/bazelbuild/bazel/issues/3450
https://stackoverflow.com/questions/45231066/linking-a-dynamic-lib-using-bazel-on-macos
https://github.com/bazelbuild/bazel/issues/4479

**Comment by alexanderlobov** [comment]
Can anybody recommend a workaround for this issue? I see several options:

1. Add `linkopts` to my `cc_binary`. In this case I should extract path to

**Comment by mhlopko** [comment]
Hi Alexander,
thank you for the report, unfortunately I cannot use your contribution. We have a full osx crosstool that needs to be fixed, but since it is generated, and the generator is not open sourced, I have to do this change internally. Thank you! I started the internal review with the fix, so it should be out soon, hopefully.

**Comment by alexanderlobov** [comment]
Do you mean that the code I fixed is generated?

**Comment by mhlopko** [comment]
No no, that is hand written, but we also have to fix osx crosstool (because that one is not patched by CppActionConfigs stuff), and that crosstool is generated. In theory we could accept your contribution and add another cl, but I'd like to keep these 2 changes together.

**Comment by alexanderlobov** [comment]
I will wait for you changes. Have closed my pull request. Thank you for your response.

**Comment by mhlopko** [comment]
Fixed in f98a7a2fedb3e714cef1038dcb85f83731150246.
